src/component/selection/DraftOffsetKey.js

```diff
@@ -25,7 +25,15 @@ const DraftOffsetKey = {
   },
 
   decode: function(offsetKey: string): DraftOffsetKeyPath {
-    const [blockKey, decoratorKey, leafKey] = offsetKey.split(KEY_DELIMITER);
+    const keyComponents = offsetKey.split(KEY_DELIMITER);
+    const blockKey = keyComponents
+      .slice(0, keyComponents.length - 2)
+      .join(KEY_DELIMITER);
+
+    const [decoratorKey, leafKey] = keyComponents.slice(
+      keyComponents.length - 2,
+    );
+
```

Comment:
I personally find this logic a bit convoluted and difficult to reason about. What about the following instead?

```js
    // Extracts the last two parts of offsetKey and captures the rest in blockKeyParts
    const [leafKey, decoratorKey, ...blockKeyParts] = offsetKey
      .split(KEY_DELIMITER)
      .reverse();

    return {
      // Recomposes the parts of blockKey after reversing them
      blockKey: blockKeyParts.reverse().join(KEY_DELIMITER),
      decoratorKey: parseInt(decoratorKey, 10),
      leafKey: parseInt(leafKey, 10),
    };
```

Reply:
Yeah, that's much nicer. Thank you!